lib/src/simulations/epicycloid_curve.dart

```diff
@@ -0,0 +1,385 @@
+import 'dart:io';
+import 'dart:math';
+import 'dart:ui';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+GlobalKey<_EpicycloidState> globalKey = GlobalKey<_EpicycloidState>();
+
+class EpicycloidCurve extends StatefulWidget {
+  @override
+  _EpicycloidCurveState createState() => _EpicycloidCurveState();
+}
+
+class _EpicycloidCurveState extends State<EpicycloidCurve> {
+  double factor = 0;
+  double total = 0;
+  bool animatefactor = false;
+  bool animatepoints = false;
+  bool animating = false;
+
+  @override
+  void initState() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+    ]);
+    super.initState();
+  }
+
+  @override
+  dispose() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+      DeviceOrientation.landscapeLeft,
+      DeviceOrientation.landscapeRight,
+    ]);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    ScreenUtil.instance = ScreenUtil(
+      width: 512.0,
+      height: 1024.0,
```

Comment:
You made this and tested on your device, you should set it to your device preferences.